File: OkHttpTest/app/src/main/java/com/xfhy/okhttptest/MainActivity.java
package com.xfhy.okhttptest;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.xfhy.myokhttp.HttpRequestListener;
import com.xfhy.myokhttp.HttpUtil;
import com.xfhy.okhttptest.test.Dictionary;
import com.xfhy.okhttptest.test.IHttpRequest;
import com.xfhy.okhttptest.test.OkHttpGet;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private static final String TAG = "MainActivity";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        HttpUtil.getRequest("http://open.iciba.com/dsapi/", null, Dictionary.class, new HttpRequestListener<Dictionary>() {
            @Override
            public void onSuccess(Dictionary dictionary) {
                Log.w(TAG, "onSuccess: " + dictionary.toString());
            }

            @Override
            public void onFailure(String errorMsg) {

            }
        });

        findViewById(R.id.btn_test).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        IHttpRequest okHttpGet = new OkHttpGet();
//        IOkHttp okHttpGet = new OkHttpPost();
        okHttpGet.request();
    }
}

File: RetrofitDemo/app/src/main/java/com/xfhy/retrofitdemo/MainActivity.kt
package com.xfhy.retrofitdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {

    private val mRetrofit by lazy {
        Retrofit.Builder()
            .baseUrl("https://wanandroid.com/")
            .addConverterFactory(GsonConverterFactory.create())   //数据解析器
            .build()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mBtnRequest.setOnClickListener {
            val iArticleApi = mRetrofit.create(IArticleApi::class.java)
            val airticlesCall = iArticleApi.getAirticles("408", "1")
            airticlesCall.enqueue(object : Callback<BaseData> {
                override fun onFailure(call: Call<BaseData>, t: Throwable) {
                    t.printStackTrace()
                    Log.e("xfhy", "请求失败")
                }

                override fun onResponse(call: Call<BaseData>, response: Response<BaseData>) {
                    val body = response.body()
                    Log.e("xfhy", "请求成功 ${body?.toString()}")
                }
            })
        }

    }
}
